### PR TITLE
fix(curriculum): better order of tests for Nutrition Step 33

### LINKED ISSUE #49227: Nutrition Label 33
### Describe the Issue

The code below generates the confusing hint

> Your `p` element should have the text `Calories`.

The actual issue is that the `span` element is in the wrong place.

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-typography-by-building-a-nutrition-label/step-33

### Your code

```html
    <div class="calories-info">
      <div class="left-container">
        <h2 class="bold small-text">Amount per serving</h2>
        <p>Calories</p>
        <span>230</span>
      </div>
    </div>
```

### Expected behavior

Clearer error message

### Screenshots

_No response_

### System

All

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jeremylt**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [ ] I have tested these changes either locally on my machine, or GitPod.

Closes #49227

### CODE REVIEW

File: curriculum/challenges/english/14-responsive-web-design-22/learn-typography-by-building-a-nutrition-label/615f5486b8fd4b71633f69b0.md [hunk 1]
```diff
@@ -29,16 +29,16 @@ Your `p` element should come after your `.small-text` element.
 assert(document.querySelector('.small-text')?.nextElementSibling?.localName === 'p');
 ```
 
-Your `p` element should have the text `Calories`.
+Your `span` element should come after your `.left-container` element.
```
Comment (jeremylt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-typography-by-building-a-nutrition-label/615f5486b8fd4b71633f69b0.md:32]: Checking the location of the span and p before checking their contents means we can check their contents however we like, making those tests less fragile.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-typography-by-building-a-nutrition-label/615f5486b8fd4b71633f69b0.md [hunk 2]
```diff
@@ -29,16 +29,16 @@ Your `p` element should come after your `.small-text` element.
...
-assert(document.querySelector('.left-container')?.lastElementChild?.innerText === 'Calories');
+assert(document.querySelector('.left-container')?.nextElementSibling?.localName ==='span');
 ```
 
-Your `span` element should come after your `.left-container` element.
+Your `p` element should have the text `Calories`.
 
 ```js
-assert(document.querySelector('.left-container')?.nextElementSibling?.localName ==='span');
+assert(document.querySelector('.left-container')?.lastElementChild?.innerText === 'Calories');
```
Comment (jeremylt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-typography-by-building-a-nutrition-label/615f5486b8fd4b71633f69b0.md:41]: We can also 'belt-and-suspenders' this so the check for the contents for p match the test for the location of p, but I still think checking the location of both *first* is less fragile
```suggestion
assert(document.querySelector('.left-container')?.nextElementSibling?.innerText === 'Calories');
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-typography-by-building-a-nutrition-label/615f5486b8fd4b71633f69b0.md [hunk 3]
```diff
@@ -29,16 +29,16 @@ Your `p` element should come after your `.small-text` element.
...
-assert(document.querySelector('.left-container')?.lastElementChild?.innerText === 'Calories');
+assert(document.querySelector('.left-container')?.nextElementSibling?.localName ==='span');
 ```
 
-Your `span` element should come after your `.left-container` element.
+Your `p` element should have the text `Calories`.
 
 ```js
-assert(document.querySelector('.left-container')?.nextElementSibling?.localName ==='span');
+assert(document.querySelector('.left-container')?.nextElementSibling?.innerText === 'Calories');
```
Comment (jeremylt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-typography-by-building-a-nutrition-label/615f5486b8fd4b71633f69b0.md:41]:
```suggestion
assert(document.querySelector('.small-text')?.nextElementSibling?.innerText === 'Calories');
```

**Review by Sboonny — APPROVED** [review]
LGTM 🎉, thank you for your contribution to the page! 👍.

**Review by gikf — APPROVED** [review]
LGTM :+1:

**Pull request merged**